### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by gaurikadiwan**
### Project Title 
Alzheimer's Disease Classification Using Vision Transformer(Task 6).
### Project Overview
In this project, I  developed a deep learning model to classify brain scans from the Alzheimer's Disease Neuroimaging  Initiative (ADNI) dataset into two categories: Alzheimer's Disease (AD) and Normal Control (NC). The model utilizes Vision Transformers (ViT) to perform image classification.
## Project Files

The project consists of several files:

1.`dataset.py:` Defines a custom dataset class for AD and NC image data, along with associated transformations and data loaders for training and testing.

2.`modules.py`: This module defines the neural network components utilized in the Vision Transformer architecture, including embeddor and the main Vision Module model ideal for image classification tasks.

3.` train.py`: Initializes, trains, and tests the Vision Module on AD and NC image data. After training, the model is saved, and training and testing metrics are visualized.

4.`predict.py`: This script loads a trained Vision Module model and uses it to predict the class of a given image. It provides a confidence score bar chart.

5.` README.md`:This file contains a detailed report about the project.

## Overall analysis
Being able to classify brain scans into categories such as Alzheimer's Disease (AD) and Normal Control (NC) can contribute to early diagnosis and intervention. I got an accuracy of 88% for 10 epochs for training and 76% for testing dataset. the maximum I could run on my device. These constraints impacted my the depth of model training, affecting the final accuracy. In addition, I gained in depth understanding of image classification, neural networks, and the practical aspects of training and evaluating machine learning models.

I would like to express my gratitude to the project team and instructors for their support and guidance throughout this project. I would highly appreciate and welcome your feedback, for the same!

Thankyou!
Gaurika Diwan
(48240983)

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
Excellent

Commit messages: Excellent, detailed.

Code:
- Uses ViT
- Well commented
- Good model design

Functionality/Performance:
- Test accuracy is 76%
- **Don't have train, val and test sets. Only have train and test sets.** Missing a validation set
- No evidence show patient-level split
- Preprocssing data correctly
- Performed data augmentation
- Well done

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no patient level splits -1, no validation set -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove model file and move README and restore repo README -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by gaurikadiwan** [comment]
Requested Changes are incorporated , removed the  model file ,moved README and restored the  repo README.

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
